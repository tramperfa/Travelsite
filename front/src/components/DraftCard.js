import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {withStyles} from 'material-ui/styles';
import Card, {CardContent, CardMedia} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import moment from 'moment';
import {graphql} from 'react-apollo';
import Edit from "material-ui-icons/Edit";
import Delete from "material-ui-icons/Delete";

//
import {draftsListQuery} from '../graphql/draft';
import {DeleteDraftMutation} from '../graphql/draft';

//
import imageTest from '../images/b.jpg';

const styles = theme => ({
  card: {
    display: 'flex',
    marginTop: 24,
    width: 620,
    height: 150
  },
  details: {
    display: 'flex',
    flexDirection: 'column'
  },
  content: {
    flex: '1 0 auto'
  },
  cover: {
    width: 220,
    height: 150
  },
  textField: {
    textDecoration: 'none'
  }
});

function DraftCard(props) {

  // class DraftCard extends React.Component {

  const handleDelete = () => {
    try {
      props.deleteDraft(props.draft._id)
    } catch (e) {
      console.log(e.graphQLErrors[0].message);
    } finally {}
  }

  // render() {
  const {classes, draft} = props;
  return (

    <div >
      <Card className={classes.card}>
        <Link className={classes.textField} to={`/edit/${draft._id}`}>
          <CardMedia
            className={classes.cover}
            image={imageTest}
            title="Live from space album cover"/>
        </Link>
        <div className={classes.details}>
          <Link className={classes.textField} to={`/edit/${draft._id}`}>
            <CardContent className={classes.content}>
              <Typography type="headline">
                {draft.title}
              </Typography>
              <Typography type="subheading" color="secondary">
                <div>
                  Last Updated: {moment(new Date(draft.lastUpdate)).utc().local().format("YYYY-MM-DD HH:mm")}
                </div>
              </Typography>
            </CardContent>
          </Link>
          <div className={classes.controls}>
            Continue Writing
            <IconButton aria-label="Edit">
              <Edit/>
            </IconButton>

            <IconButton aria-label="Delete" onClick={handleDelete}>
              <Delete/>
            </IconButton>

          </div>
        </div>

      </Card>
    </div>
  );
  // }
}

DraftCard.propTypes = {
  classes: PropTypes.object.isRequired,
  //theme: PropTypes.object.isRequired,
  deleteDraft: PropTypes.func.isRequired,
  draft: PropTypes.object.isRequired
};

export const WithDelete = graphql(DeleteDraftMutation, {
  props: ({mutate}) => ({
    deleteDraft: (draftID) => mutate({
      variables: {
        draftID: draftID
      },
      refetchQueries: [
        {
          query: draftsListQuery
        }
      ]
    })
  })
})

export default WithDelete(withStyles(styles, {withTheme: true})(DraftCard))
