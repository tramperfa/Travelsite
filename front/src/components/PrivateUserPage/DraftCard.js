import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import format from 'date-fns/format'
import {compose} from 'recompose';

//
import {withStyles} from 'material-ui/styles';
import Card, {CardContent, CardMedia} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import Edit from "material-ui-icons/Edit";
import Delete from "material-ui-icons/Delete";

//
import {WithDeleteDraftMutation} from '../../graphql/draft';
import ErrorComp from '../../lib/ErrorComp';
import {error, onError} from '../../lib/utils';
import CONSTS from '../../lib/consts';

//
import defaultbrowserCoverImage from '../../images/browserCoverImage.png';

export const styles = theme => ({
	card: {
		display: 'flex',
		marginTop: 24,
		width: 620,
		height: 150
	},
	details: {
		display: 'flex',
		flexDirection: 'column'
	},
	content: {
		flex: '1 0 auto'
	},
	cover: {
		width: 220,
		height: 150
	},
	textField: {
		textDecoration: 'none'
	}
});

class DraftCard extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			error: error
		}
	}

	handleDelete = () => {
		this.props.deleteDraft(this.props.draft._id).then(() => {
			//this.setState({deleteRedirect: true})
		}).catch((err) => {
			this.setState(onError(err))
		})
	}

	render() {
		const {classes, draft} = this.props;
		return (
			<div>
				<ErrorComp error={this.state.error}/>
				<Card className={classes.card}>
					<Link className={classes.textField} to={`/my/edit/${draft._id}`}>
						<CardMedia
							className={classes.cover}
							image={draft.coverImage
								? CONSTS.BUCKET_NAME + draft.coverImage.browserCoverImage.filename
								: defaultbrowserCoverImage}/>
					</Link>
					<div className={classes.details}>
						<Link className={classes.textField} to={`/my/edit/${draft._id}`}>
							<CardContent className={classes.content}>
								<Typography type="headline">
									{draft.title}
								</Typography>
								<Typography type="subheading" color="secondary">
									<div>
										Last Updated: {format(new Date(draft.lastUpdate), "YYYY-MM-DD HH:mm")}
									</div>
								</Typography>
							</CardContent>
						</Link>
						<div className={classes.controls}>
							Continue Writing
							<IconButton aria-label="Edit">
								<Edit/>
							</IconButton>
							<IconButton aria-label="Delete" onClick={this.handleDelete}>
								<Delete/>
							</IconButton>
						</div>
					</div>
				</Card>
			</div>
		);

	}

}

DraftCard.propTypes = {
	classes: PropTypes.object.isRequired,
	deleteDraft: PropTypes.func.isRequired,
	draft: PropTypes.object.isRequired
};

export default compose(WithDeleteDraftMutation, withStyles(styles))(DraftCard)
