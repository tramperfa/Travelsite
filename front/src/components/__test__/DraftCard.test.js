import React from 'react';
import renderer from 'react-test-renderer';
// Note: test renderer must be required after react-native. i
import {DraftCard, style} from '../DraftCard';
import TheDraftCard from '../DraftCard';
import {shallow, mount} from 'enzyme';
import {MemoryRouter, StaticRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json'

// test('shallow renders correctly', () => { 	const wrapper = shallow(
// <MemoryRouter> <TheDraftCard deleteDraft={() => {}} draft={{}} classes={{}}/>
// </MemoryRouter>
//
// ); 	const render = wrapper.dive() 	console.log(JSON.stringify(render));
// render.find('IconButton').forEach(child => { 		console.log("SIMULATED");
// child.simulate('click'); 	}); 	expect(wrapper).toMatchSnapshot(); });

test('Tree renders correctly', () => {
	const tree = renderer.create(
		<StaticRouter context={{}}>
			<DraftCard
				deleteDraft={() => {}}
				draft={{
					_id: 'test000'
				}}
				classes={style}/>
		</StaticRouter>
	).toJSON();
	expect(tree).toMatchSnapshot();
});

test('Mount renders correctly', () => {
	const wrapper = mount(
		<StaticRouter context={{}}>
			<DraftCard deleteDraft={() => {}} draft={{}} classes={{}}/>
		</StaticRouter>

	);

	console.log(JSON.stringify(wrapper));
	wrapper.find('button').forEach(child => {
		console.log("SIMULATED");
		child.simulate('click');
	});
	//expect(handleDelete.calledOnce).toBe(true);
	expect(toJson(wrapper)).toMatchSnapshot();
});
