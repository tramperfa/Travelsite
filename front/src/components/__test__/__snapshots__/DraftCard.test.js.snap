// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount renders correctly 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      classes={Object {}}
      deleteDraft={[Function]}
      draft={Object {}}
    >
      <div>
        <Card
          raised={false}
        >
          <withStyles(Paper)
            elevation={2}
          >
            <Paper
              classes={
                Object {
                  "root": "MuiPaper-root-1",
                  "rounded": "MuiPaper-rounded-2",
                  "shadow0": "MuiPaper-shadow0-3",
                  "shadow1": "MuiPaper-shadow1-4",
                  "shadow10": "MuiPaper-shadow10-13",
                  "shadow11": "MuiPaper-shadow11-14",
                  "shadow12": "MuiPaper-shadow12-15",
                  "shadow13": "MuiPaper-shadow13-16",
                  "shadow14": "MuiPaper-shadow14-17",
                  "shadow15": "MuiPaper-shadow15-18",
                  "shadow16": "MuiPaper-shadow16-19",
                  "shadow17": "MuiPaper-shadow17-20",
                  "shadow18": "MuiPaper-shadow18-21",
                  "shadow19": "MuiPaper-shadow19-22",
                  "shadow2": "MuiPaper-shadow2-5",
                  "shadow20": "MuiPaper-shadow20-23",
                  "shadow21": "MuiPaper-shadow21-24",
                  "shadow22": "MuiPaper-shadow22-25",
                  "shadow23": "MuiPaper-shadow23-26",
                  "shadow24": "MuiPaper-shadow24-27",
                  "shadow3": "MuiPaper-shadow3-6",
                  "shadow4": "MuiPaper-shadow4-7",
                  "shadow5": "MuiPaper-shadow5-8",
                  "shadow6": "MuiPaper-shadow6-9",
                  "shadow7": "MuiPaper-shadow7-10",
                  "shadow8": "MuiPaper-shadow8-11",
                  "shadow9": "MuiPaper-shadow9-12",
                }
              }
              component="div"
              elevation={2}
              square={false}
            >
              <div
                className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
              >
                <Link
                  replace={false}
                  to="/edit/undefined"
                >
                  <a
                    href="/edit/undefined"
                    onClick={[Function]}
                  >
                    <withStyles(CardMedia)
                      image="b.jpg"
                      title="Live from space album cover"
                    >
                      <CardMedia
                        classes={
                          Object {
                            "root": "MuiCardMedia-root-28",
                            "rootMedia": "MuiCardMedia-rootMedia-29",
                          }
                        }
                        component="div"
                        image="b.jpg"
                        title="Live from space album cover"
                      >
                        <div
                          className="MuiCardMedia-root-28"
                          style={
                            Object {
                              "backgroundImage": "url(b.jpg)",
                            }
                          }
                          title="Live from space album cover"
                        />
                      </CardMedia>
                    </withStyles(CardMedia)>
                  </a>
                </Link>
                <div>
                  <Link
                    replace={false}
                    to="/edit/undefined"
                  >
                    <a
                      href="/edit/undefined"
                      onClick={[Function]}
                    >
                      <withStyles(CardContent)>
                        <CardContent
                          classes={
                            Object {
                              "root": "MuiCardContent-root-30",
                            }
                          }
                        >
                          <div
                            className="MuiCardContent-root-30"
                          >
                            <withStyles(Typography)
                              type="headline"
                            >
                              <Typography
                                align="inherit"
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter-44",
                                    "alignJustify": "MuiTypography-alignJustify-46",
                                    "alignLeft": "MuiTypography-alignLeft-43",
                                    "alignRight": "MuiTypography-alignRight-45",
                                    "body1": "MuiTypography-body1-40",
                                    "body2": "MuiTypography-body2-39",
                                    "button": "MuiTypography-button-42",
                                    "caption": "MuiTypography-caption-41",
                                    "colorAccent": "MuiTypography-colorAccent-53",
                                    "colorError": "MuiTypography-colorError-54",
                                    "colorInherit": "MuiTypography-colorInherit-50",
                                    "colorPrimary": "MuiTypography-colorPrimary-51",
                                    "colorSecondary": "MuiTypography-colorSecondary-52",
                                    "display1": "MuiTypography-display1-35",
                                    "display2": "MuiTypography-display2-34",
                                    "display3": "MuiTypography-display3-33",
                                    "display4": "MuiTypography-display4-32",
                                    "gutterBottom": "MuiTypography-gutterBottom-48",
                                    "headline": "MuiTypography-headline-36",
                                    "noWrap": "MuiTypography-noWrap-47",
                                    "paragraph": "MuiTypography-paragraph-49",
                                    "root": "MuiTypography-root-31",
                                    "subheading": "MuiTypography-subheading-38",
                                    "title": "MuiTypography-title-37",
                                  }
                                }
                                color="default"
                                gutterBottom={false}
                                headlineMapping={
                                  Object {
                                    "body1": "p",
                                    "body2": "aside",
                                    "display1": "h1",
                                    "display2": "h1",
                                    "display3": "h1",
                                    "display4": "h1",
                                    "headline": "h1",
                                    "subheading": "h3",
                                    "title": "h2",
                                  }
                                }
                                noWrap={false}
                                paragraph={false}
                                type="headline"
                              >
                                <h1
                                  className="MuiTypography-root-31 MuiTypography-headline-36"
                                />
                              </Typography>
                            </withStyles(Typography)>
                            <withStyles(Typography)
                              color="secondary"
                              type="subheading"
                            >
                              <Typography
                                align="inherit"
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter-44",
                                    "alignJustify": "MuiTypography-alignJustify-46",
                                    "alignLeft": "MuiTypography-alignLeft-43",
                                    "alignRight": "MuiTypography-alignRight-45",
                                    "body1": "MuiTypography-body1-40",
                                    "body2": "MuiTypography-body2-39",
                                    "button": "MuiTypography-button-42",
                                    "caption": "MuiTypography-caption-41",
                                    "colorAccent": "MuiTypography-colorAccent-53",
                                    "colorError": "MuiTypography-colorError-54",
                                    "colorInherit": "MuiTypography-colorInherit-50",
                                    "colorPrimary": "MuiTypography-colorPrimary-51",
                                    "colorSecondary": "MuiTypography-colorSecondary-52",
                                    "display1": "MuiTypography-display1-35",
                                    "display2": "MuiTypography-display2-34",
                                    "display3": "MuiTypography-display3-33",
                                    "display4": "MuiTypography-display4-32",
                                    "gutterBottom": "MuiTypography-gutterBottom-48",
                                    "headline": "MuiTypography-headline-36",
                                    "noWrap": "MuiTypography-noWrap-47",
                                    "paragraph": "MuiTypography-paragraph-49",
                                    "root": "MuiTypography-root-31",
                                    "subheading": "MuiTypography-subheading-38",
                                    "title": "MuiTypography-title-37",
                                  }
                                }
                                color="secondary"
                                gutterBottom={false}
                                headlineMapping={
                                  Object {
                                    "body1": "p",
                                    "body2": "aside",
                                    "display1": "h1",
                                    "display2": "h1",
                                    "display3": "h1",
                                    "display4": "h1",
                                    "headline": "h1",
                                    "subheading": "h3",
                                    "title": "h2",
                                  }
                                }
                                noWrap={false}
                                paragraph={false}
                                type="subheading"
                              >
                                <h3
                                  className="MuiTypography-root-31 MuiTypography-subheading-38 MuiTypography-colorSecondary-52"
                                >
                                  <div>
                                    Last Updated: 
                                    Invalid date
                                  </div>
                                </h3>
                              </Typography>
                            </withStyles(Typography)>
                          </div>
                        </CardContent>
                      </withStyles(CardContent)>
                    </a>
                  </Link>
                  <div>
                    Continue Writing
                    <withStyles(IconButton)
                      aria-label="Edit"
                    >
                      <IconButton
                        aria-label="Edit"
                        classes={
                          Object {
                            "colorAccent": "MuiIconButton-colorAccent-56",
                            "colorContrast": "MuiIconButton-colorContrast-57",
                            "colorInherit": "MuiIconButton-colorInherit-59",
                            "colorPrimary": "MuiIconButton-colorPrimary-58",
                            "disabled": "MuiIconButton-disabled-60",
                            "icon": "MuiIconButton-icon-62",
                            "keyboardFocused": "MuiIconButton-keyboardFocused-63",
                            "label": "MuiIconButton-label-61",
                            "root": "MuiIconButton-root-55",
                          }
                        }
                        color="default"
                        disableRipple={false}
                        disabled={false}
                      >
                        <withStyles(ButtonBase)
                          aria-label="Edit"
                          centerRipple={true}
                          className="MuiIconButton-root-55"
                          disableRipple={false}
                          disabled={false}
                          keyboardFocusedClassName="MuiIconButton-keyboardFocused-63"
                        >
                          <ButtonBase
                            aria-label="Edit"
                            centerRipple={true}
                            className="MuiIconButton-root-55"
                            classes={
                              Object {
                                "disabled": "MuiButtonBase-disabled-65",
                                "root": "MuiButtonBase-root-64",
                              }
                            }
                            disableRipple={false}
                            disabled={false}
                            focusRipple={false}
                            keyboardFocusedClassName="MuiIconButton-keyboardFocused-63"
                            tabIndex={0}
                            type="button"
                          >
                            <button
                              aria-label="Edit"
                              className="MuiButtonBase-root-64 MuiIconButton-root-55"
                              disabled={false}
                              onBlur={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label-61"
                              >
                                <pure(Edit)
                                  className="MuiIconButton-icon-62"
                                  key=".0"
                                >
                                  <Edit
                                    className="MuiIconButton-icon-62"
                                  >
                                    <withStyles(SvgIcon)
                                      className="MuiIconButton-icon-62"
                                    >
                                      <SvgIcon
                                        className="MuiIconButton-icon-62"
                                        classes={
                                          Object {
                                            "root": "MuiSvgIcon-root-66",
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className="MuiSvgIcon-root-66 MuiIconButton-icon-62"
                                          focusable="false"
                                          viewBox="0 0 24 24"
                                        >
                                          <path
                                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                                          />
                                        </svg>
                                      </SvgIcon>
                                    </withStyles(SvgIcon)>
                                  </Edit>
                                </pure(Edit)>
                              </span>
                              <withStyles(TouchRipple)
                                center={true}
                                innerRef={[Function]}
                              >
                                <TouchRipple
                                  center={true}
                                  classes={
                                    Object {
                                      "ripple": "MuiTouchRipple-ripple-71",
                                      "rippleFast": "MuiTouchRipple-rippleFast-73",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible-72",
                                      "root": "MuiTouchRipple-root-67",
                                      "wrapper": "MuiTouchRipple-wrapper-68",
                                      "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-69",
                                      "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-70",
                                    }
                                  }
                                >
                                  <TransitionGroup
                                    childFactory={[Function]}
                                    className="MuiTouchRipple-root-67"
                                    component="span"
                                    enter={true}
                                    exit={true}
                                  >
                                    <span
                                      className="MuiTouchRipple-root-67"
                                    />
                                  </TransitionGroup>
                                </TouchRipple>
                              </withStyles(TouchRipple)>
                            </button>
                          </ButtonBase>
                        </withStyles(ButtonBase)>
                      </IconButton>
                    </withStyles(IconButton)>
                    <withStyles(IconButton)
                      aria-label="Delete"
                      onClick={[Function]}
                    >
                      <IconButton
                        aria-label="Delete"
                        classes={
                          Object {
                            "colorAccent": "MuiIconButton-colorAccent-56",
                            "colorContrast": "MuiIconButton-colorContrast-57",
                            "colorInherit": "MuiIconButton-colorInherit-59",
                            "colorPrimary": "MuiIconButton-colorPrimary-58",
                            "disabled": "MuiIconButton-disabled-60",
                            "icon": "MuiIconButton-icon-62",
                            "keyboardFocused": "MuiIconButton-keyboardFocused-63",
                            "label": "MuiIconButton-label-61",
                            "root": "MuiIconButton-root-55",
                          }
                        }
                        color="default"
                        disableRipple={false}
                        disabled={false}
                        onClick={[Function]}
                      >
                        <withStyles(ButtonBase)
                          aria-label="Delete"
                          centerRipple={true}
                          className="MuiIconButton-root-55"
                          disableRipple={false}
                          disabled={false}
                          keyboardFocusedClassName="MuiIconButton-keyboardFocused-63"
                          onClick={[Function]}
                        >
                          <ButtonBase
                            aria-label="Delete"
                            centerRipple={true}
                            className="MuiIconButton-root-55"
                            classes={
                              Object {
                                "disabled": "MuiButtonBase-disabled-65",
                                "root": "MuiButtonBase-root-64",
                              }
                            }
                            disableRipple={false}
                            disabled={false}
                            focusRipple={false}
                            keyboardFocusedClassName="MuiIconButton-keyboardFocused-63"
                            onClick={[Function]}
                            tabIndex={0}
                            type="button"
                          >
                            <button
                              aria-label="Delete"
                              className="MuiButtonBase-root-64 MuiIconButton-root-55"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label-61"
                              >
                                <pure(Delete)
                                  className="MuiIconButton-icon-62"
                                  key=".0"
                                >
                                  <Delete
                                    className="MuiIconButton-icon-62"
                                  >
                                    <withStyles(SvgIcon)
                                      className="MuiIconButton-icon-62"
                                    >
                                      <SvgIcon
                                        className="MuiIconButton-icon-62"
                                        classes={
                                          Object {
                                            "root": "MuiSvgIcon-root-66",
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className="MuiSvgIcon-root-66 MuiIconButton-icon-62"
                                          focusable="false"
                                          viewBox="0 0 24 24"
                                        >
                                          <path
                                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                                          />
                                        </svg>
                                      </SvgIcon>
                                    </withStyles(SvgIcon)>
                                  </Delete>
                                </pure(Delete)>
                              </span>
                              <withStyles(TouchRipple)
                                center={true}
                                innerRef={[Function]}
                              >
                                <TouchRipple
                                  center={true}
                                  classes={
                                    Object {
                                      "ripple": "MuiTouchRipple-ripple-71",
                                      "rippleFast": "MuiTouchRipple-rippleFast-73",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible-72",
                                      "root": "MuiTouchRipple-root-67",
                                      "wrapper": "MuiTouchRipple-wrapper-68",
                                      "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-69",
                                      "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-70",
                                    }
                                  }
                                >
                                  <TransitionGroup
                                    childFactory={[Function]}
                                    className="MuiTouchRipple-root-67"
                                    component="span"
                                    enter={true}
                                    exit={true}
                                  >
                                    <span
                                      className="MuiTouchRipple-root-67"
                                    />
                                  </TransitionGroup>
                                </TouchRipple>
                              </withStyles(TouchRipple)>
                            </button>
                          </ButtonBase>
                        </withStyles(ButtonBase)>
                      </IconButton>
                    </withStyles(IconButton)>
                  </div>
                </div>
              </div>
            </Paper>
          </withStyles(Paper)>
        </Card>
      </div>
    </Component>
  </Router>
</StaticRouter>
`;

exports[`Tree renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
  >
    <a
      className={undefined}
      href="/edit/test000"
      onClick={[Function]}
    >
      <div
        className="MuiCardMedia-root-28"
        src={undefined}
        style={
          Object {
            "backgroundImage": "url(b.jpg)",
          }
        }
        title="Live from space album cover"
      />
    </a>
    <div
      className={undefined}
    >
      <a
        className={undefined}
        href="/edit/test000"
        onClick={[Function]}
      >
        <div
          className="MuiCardContent-root-30"
        >
          <h1
            className="MuiTypography-root-31 MuiTypography-headline-36"
          />
          <h3
            className="MuiTypography-root-31 MuiTypography-subheading-38 MuiTypography-colorSecondary-52"
          >
            <div>
              Last Updated: 
              Invalid date
            </div>
          </h3>
        </div>
      </a>
      <div
        className={undefined}
      >
        Continue Writing
        <button
          aria-label="Edit"
          className="MuiButtonBase-root-64 MuiIconButton-root-55"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-61"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-66 MuiIconButton-icon-62"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-67"
          />
        </button>
        <button
          aria-label="Delete"
          className="MuiButtonBase-root-64 MuiIconButton-root-55"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-61"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-66 MuiIconButton-icon-62"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-67"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
